File: Easy/Merge Two Sorted Lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        if (list1 == null && list2 == null) return null;
        if (list1 == null) return list2;
        if (list2 == null) return list1;
        
        ListNode rs = new ListNode(), temp = rs;
        
        ListNode t1 = list1, t2 = list2;
        
        while (t1 != null && t2 != null) {
            if (t1.val >= t2.val) {
                temp.next = t2;
                t2 = t2.next;
            } else {
                temp.next = t1;
                t1 = t1.next;
            }
            
            temp = temp.next;
        }
        
        if (t1 != null) temp.next = t1;
        if (t2 != null) temp.next = t2;
        
        return rs.next;
    }
}

File: Easy/Merge Two Sorted Lists.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* mergeTwoLists(struct ListNode* l1, struct ListNode* l2){
    if(l1 == NULL && l2 == NULL) return NULL;
    struct ListNode* rNode = (struct ListNode*)malloc(sizeof(struct ListNode)), *ltemp = l1, *rtemp = l2;
    rNode->next = NULL;
    struct ListNode* head = rNode;

    while(ltemp != NULL && rtemp != NULL){
        struct ListNode* temp = (struct ListNode*)malloc(sizeof(struct ListNode));
        temp->next = NULL;
        if(ltemp->val <= rtemp->val){
            temp->val = ltemp->val;
            ltemp = ltemp->next;
        }
        else{
            temp->val = rtemp->val;
            rtemp = rtemp->next;
        }
        rNode->next = temp;
        rNode = rNode->next;
    }
    if(rtemp != NULL){
        while(rtemp != NULL){
            struct ListNode* temp = (struct ListNode*)malloc(sizeof(struct ListNode));
            temp->val = rtemp->val;
            temp->next = NULL;
            rNode->next = temp;
            rNode = rNode->next;
            rtemp = rtemp->next;
        }
    }
    if(ltemp != NULL){
        while(ltemp != NULL){
            struct ListNode* temp = (struct ListNode*)malloc(sizeof(struct ListNode));
            temp->val = ltemp->val;
            temp->next = NULL;
            rNode->next = temp;
            rNode = rNode->next;
            ltemp = ltemp->next;
        }
    }
    
    return head->next;
}